lib/Internal/functions.php

```diff
@@ -108,6 +108,12 @@ function bootServer(PsrLogger $logger, Console $console): \Generator {
         $hosts = [$hosts];
     }
 
+    if (empty($hosts)) {
+        throw new \Error(
+            "Config file at $configFile did not return any hosts"
+        );
+    }
```

Comment:
@bwoebi Should we remove the `if ($hosts === false)` check above then?

Reply:
include returns false if the file could not be loaded at all. This check here is throwing when the return value of the file is unexpected.

Reply:
Should probably just be `require` then? As it might also return `false` (in theory)?

Reply:
require will throw an \Error, yes, but it won't tell you that the file trying to be included was the indeed config file and not some random internal file (i.e. a bug on our side).

Reply:
Uhm, the error tells you the file name?

Reply:
IMHO it would indeed be clearer if the error message specifies that it was a config file inclusion that failed (as opposed to some random internal file). Differentiating between the config file returning `false` and not existing, would also be nice. To get both these things working correctly, you could check if the file exists before `require`ing it.